pkg/events/event_channel.go

```diff
@@ -0,0 +1,74 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package events
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/apache/incubator-yunikorn-core/pkg/log"
+)
+
+// TODO should configure the size of the channel
+const eventChannelSize = 100000
+
+// wrapping the channel into a struct - so that the underlying implementation can be changed
+type eventChannel struct {
+	events       chan Event
```

Comment:
this is a bit confusing to me.
if we are adding R/W lock to the channel, we do not need to use the channel
a fixed size queue with add/pop function seems to be sufficient for this

Reply:
Actually there is no queue implementation besides channel in Golang as far I understood. The lock is needed to protect the diagCounter - in a non-protected `AddEvent` I think this line is non atomic which can cause issues:
```
ec.diagCounter += 1
```